File: internal/store/postgres/organization.go
package postgres

import (
	"encoding/json"
	"time"

	"database/sql"

	"github.com/raystack/frontier/core/organization"
)

type Organization struct {
	ID        string         `db:"id"`
	Name      string         `db:"name"`
	Title     sql.NullString `db:"title"`
	Metadata  []byte         `db:"metadata"`
	State     sql.NullString `db:"state"`
	CreatedAt time.Time      `db:"created_at"`
	UpdatedAt time.Time      `db:"updated_at"`
	DeletedAt sql.NullTime   `db:"deleted_at"`
}

func (from Organization) transformToOrg() (organization.Organization, error) {
	var unmarshalledMetadata map[string]any
	if len(from.Metadata) > 0 {
		if err := json.Unmarshal(from.Metadata, &unmarshalledMetadata); err != nil {
			return organization.Organization{}, err
		}
	}

	return organization.Organization{
		ID:        from.ID,
		Name:      from.Name,
		Title:     from.Title.String,
		Metadata:  unmarshalledMetadata,
		State:     organization.State(from.State.String),
		CreatedAt: from.CreatedAt,
		UpdatedAt: from.UpdatedAt,
	}, nil
}

File: sdks/js/packages/core/shared/utils/organization.ts
import dayjs from 'dayjs';
import { timestampToDate } from './time';
import type { V1Beta1Organization } from '../../api-client';
import type { Organization } from '@raystack/proton/frontier';

/**
 * Transform ConnectRPC Organization to V1Beta1Organization
 * Converts proto timestamp fields to JSON string format
 */
export function transformConnectOrgToV1Beta1(
  connectOrg: Organization
): V1Beta1Organization {
  if (!connectOrg) return connectOrg as V1Beta1Organization;

  const createdAtString = connectOrg.createdAt
    ? dayjs(timestampToDate(connectOrg.createdAt)).toISOString()
    : undefined;

  const updatedAtString = connectOrg.updatedAt
    ? dayjs(timestampToDate(connectOrg.updatedAt)).toISOString()
    : undefined;

  return {
    id: connectOrg.id,
    name: connectOrg.name,
    title: connectOrg.title,
    metadata: connectOrg.metadata as unknown as object,
    state: connectOrg.state,
    avatar: connectOrg.avatar,
    created_at: createdAtString,
    updated_at: updatedAtString
  };
}
